### DISCUSSION TITLE
How do I make a single row noneditable/read only?

### DISCUSSION THREAD

**Discussion by islandnuge** [Power-Components/livewire-powergrid#856] [Q&A] [answered]
I'm looking to make some rows of my datagrid table entirely read-only or disabled (depending on the users's state).

I'm considering using CSS for this, but am looking to ensure that buttons and editable states are not visible underneath - is this possible?

**Comment by dansysanalyst** [comment, marked as answer]
Hi @islandnuge,

Currently, you can use[ Action Rules](https://livewire-powergrid.com/table/action-rules.html#action-rules) to control the rows.

It might look something like this: 

```php
final class MyTable extends PowerGridComponent
{
//...
 public function actionRules(): array
    {
        return [
              Rule::rows()
                ->when(fn ($user) => $user->isActive() === false)
                ->disable(),
                ->hide(),
                ->hideToggleable(),
                ->setAttribute('class', 'bg-gray-50'),

    }
```

There is no rule to disable editOnClick, but, we welcome a pull request for this feature.
